GAE/src/org/akvo/flow/api/app/DataPointServlet.java

```diff
@@ -87,7 +79,11 @@ protected RestResponse handleRequest(RestRequest req) throws Exception {
         }
 
         res = convertToResponse(dpList, dpReq.getSurveyId());
-        res.setCursor(BaseDAO.getCursor(dpList));
+        String cursor = BaseDAO.getCursor(dpList);
+        if (cursor == null && dpList.size() > 0) {
+            cursor = dpList.size() + "";
+        }
+        res.setCursor(cursor);
```

Comment:
If we follow the current behaviour of the app, it will send requests to the backend until the returned results == 0.

So take the case of 34 assigned datapoints. 
- First query = 30 datapoints and cursor=30
- second query = 4 datapoints and cursor=(4 or 34?)
- third query = 0 datapoints and cursor=(?)

Reply:
cursor would be null, I need to check if it could be an issue

Reply:
I checked and having a null cursor is not an issue. I think in this case it makes sense. User downloads all datapoints assigned to him and cursor is reset to null. Datapoints are assigned/modified. User pressed download -> download restarts from 0 (we have no way to tell if the datapoints have been modified recently because we don't have a "real" cursor)